src/main/java/io/codelex/flightplanner/Service/FlightServiceDatabase.java

```diff
@@ -0,0 +1,104 @@
+package io.codelex.flightplanner.Service;
+
+import io.codelex.flightplanner.domain.*;
+import io.codelex.flightplanner.repository.AirportsRepositoryDatabase;
+import io.codelex.flightplanner.repository.FlightRepositoryDatabase;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.parameters.P;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static io.codelex.flightplanner.utils.Utilities.DATE_TIME_FORMATTER;
+
+@Service
+@ConditionalOnProperty(prefix = "flightapp", name = "appmode", havingValue = "database")
+public class FlightServiceDatabase implements FlightsService {
+
+    private final FlightRepositoryDatabase flightRepo;
+    private final AirportsRepositoryDatabase airportRepo;
+
+    @Autowired
+    public FlightServiceDatabase(FlightRepositoryDatabase flightRepo, AirportsRepositoryDatabase airportRepo) {
+        this.flightRepo = flightRepo;
+        this.airportRepo = airportRepo;
+    }
+
+    @Override
+    public void clear() {
+        flightRepo.deleteAll();
+        airportRepo.deleteAll();
+    }
+
+    @Override
+    public Flight addFlight(AddFlightRequest addFlightRequest) {
+        Flight flight = addFlightRequest.changeToFlight();
+        if (flightRepo.findAll().stream().anyMatch(flight1 -> flight1.equals(flight))) {
```

Comment:
Vot šitā nevajadzētu darīt:
`flightRepo.findAll()`
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries